enhancements/network/bgp-ovn-kubernetes.md

```diff
@@ -0,0 +1,366 @@
+
+LoadBalancer and External IP are supported by MetalLB. This feature will handle advertising cluster IPs for services.
+Note, the usefulness of advertising ClusterIP may be limited as it will be advertised by multiple nodes (similar to
+anycast) and may not work well with stateful connections.
+
+##### Egress Service
+
+Full support.
+
+##### Egress Firewall
+
+Full support.
+
+##### Egress QoS
+
+Full Support.
+
+
+##### Network Policy/ANP
+
+Full Support.
+
+##### Direct Pod Ingress
+
+Direct pod ingress is already enabled for the default cluster network. With direct pod ingress, any external entity can
+talk to a pod IP if it sends the packet to the node where the pod lives. Previously, support for direct pod ingress on
+user-defined networks was not supported. RouteAdvertisements with this enhancement may select user-defined networks and
+enable pod network advertisements. Therefore, it only makes sense to also accept direct pod ingress for these
+user-defined networks as well, especially since we know the selected subnets will not overlap in IP address space.
+
+Modifications will need to be made so that for packets arriving on br-ex, flows are added to steer the selected pod
+subnets to the right OVN GR patch port. Additionally, modifications will need to be made in the kernel routing table to
+leak routes to the pod subnets from each user-defined VRF into the default VRF routing table.
+
+#### Deployment Considerations
+
+##### MetalLB
+
+MetalLB is supported today as a day 2 add-on operator. MetalLB supports FRR mode, and has recently added support for
+FRR-K8S mode. When MetalLB is deployed in FRR-K8S mode, it handles instantiating FRR-K8S. For the purposes of
+OVN-Kubernetes integration, we need this as a day 0 function. For pods relying on the underlay and not using an overlay
+encapsulation, the BGP routes will need to be learned by the time the pods come up.
+
+In order to achieve day 0 functionality, FRR-K8S will be launched by CNO as a host networked pod on each node (where
+applicable). MetalLB will be modified so that it can launch in FRR-K8S mode, without deploying FRR-K8S, and integrate
+with the one launched by CNO.
+
+##### Bare Metal
+
+Bare Metal deployments expose a virtual IP (VIP) using keepalived. This VIP is moved between control-plane nodes, and
+the implementation assumes the master nodes are on the same layer 2 network. With BGP integration, this may no longer be
+the case. Therefore, we may need changes to keepalived scripts in order to add FRR configuration for routes to the VIP
+when the VIP goes up/down on a node, so that the VIP is announced correctly over the BGP network.\
+
+## Test Plan
+
+* E2E upstream with a framework to simulate a routed spine and leaf topology with integration using OVN Kubernetes.
+* Testing using transport none for some networks, and Geneve for non-BGP enabled networks.
+* Telco team downstream testing to cover BGP functionality including MEG, Egress IP, Egress QoS, etc.
```

Comment:
this needs to be agreed upon I think (not from me I mean :P)

Reply:
Telco Engineering has already planned 4.17 work. No spare QE capacity.

We can always adjust priorities, but I strongly recommend seeing the design phase through and good progress on implementation before considering changes. This will allow us to make informed decisions about potential shifts later.

Reply:
I'll update this and leave out the specific team responsibility.